pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -334,5 +334,47 @@ type fileDescription struct {
 	OpenFlag uint32
 }
 
+func (fd *fileDescription) dentry() *kernfs.Dentry {
+	return fd.vfsfd.Dentry().Impl().(*kernfs.Dentry)
+}
+
+func (fd *fileDescription) inode() *Inode {
+	return fd.dentry().Inode().(*Inode)
+}
+
+func (fd *fileDescription) statusFlags() uint32 {
+	return fd.vfsfd.StatusFlags()
+}
+
 // Release implements vfs.FileDescriptionImpl.Release.
 func (fd *fileDescription) Release() {}
+
+// OnClose implements vfs.FileDescriptionImpl.OnClose.
+func (fd *fileDescription) OnClose(ctx context.Context) error {
+	// only send FUSE_RELEASE when there is no reference to current fd.
+	// this makes Onclose perform similar logic as Release(), since
+	// Release() doesn't have ctx, use OnClose instead.
+	if fd.vfsfd.Refs() != 0 {
+		return nil
+	}
+
+	in := linux.FUSEReleaseIn{
+		Fh:    fd.Fh,
+		Flags: fd.statusFlags(),
+	}
+	// TODO(gvisor.dev/issue/3245): add logic when we support file lock owner.
+	var opcode linux.FUSEOpcode
+	if fd.inode().Mode().IsDir() {
+		opcode = linux.FUSE_RELEASEDIR
+	} else {
+		opcode = linux.FUSE_RELEASE
+	}
+	conn := fd.inode().fs.conn
+	if !conn.NoOpen {
+		// ignoring error is analogous to Linux's behavior.
+		req, _ := conn.NewRequest(auth.CredentialsFromContext(ctx), uint32(kernel.TaskFromContext(ctx).ThreadID()), fd.inode().Ino(), opcode, &in)
+		// FUSE server doesn't reply to this call doesn't
```

Comment:
"FUSE server doesn't reply to this call doesn't" doesn't sound like a complete sentence to me?